File: uploads/docsnsi/comparaison_langage/factorielle.c
#include<stdio.h>
#include<stdlib.h>

int factorielle ( int n ) {
  if ( n == 0 || n == 1 ) {
    return 1;
  }
  else {
    int result = 1;
    int i = 0;
    while (i < n) {
      i = i + 1;
      result = result * i;
    }
    return result;
  }
}

int main(int argc, char *argv[] ) {
  int n = atoi(argv[1]);
  printf("%d", factorielle(n) );
} 
  

File: uploads/docsnsi/comparaison_langage/factorielle.py
def factorielle(n):
    if n == 0 or n == 1:
        return 1
    else:
        result = 1
        i = 0
        while i < n:
            i = i + 1
            result = result * i
        return result

factorielle(5)

if __name__ == '__main__':
   import sys
   n = int(sys.argv[1])
   print( str(factorielle(n)) )